### PR TITLE
External contract call interfaces

### PR DESCRIPTION
**Pull request by jacqueswww**
### What I did
Fixes #1214 #885 (phase2) to allow external interfaces to be used to make calls.
Fixes #1215.

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by fubuloubu** [comment]
This breaking change should derive from a VIP discussion so it's more obvious where it stems from

**Comment by jacqueswww** [comment]
We discussed these on call, 28th January IIRC

**Comment by fubuloubu** [comment]
Issues, or it didn't happen :stuck_out_tongue:

**Comment by jacqueswww** [comment]
I'll amend the original VIP.

**Comment by jacqueswww** [comment]
@fubuloubu I see I did actually note phase 2 on the VIP: https://github.com/ethereum/vyper/issues/885#issuecomment-458183821

**Comment by fubuloubu** [comment]
I still don't think the specific change `address(Interface)` was discussed outside of gitter, so I would wait until the next meeting to finalize/accept this PR

**Comment by jacqueswww** [comment]
@fubuloubu  :+1:  https://github.com/ethereum/vyper/issues/1251 knock yourself out.

**Comment by jacqueswww** [comment]
@fubuloubu

**Comment by jacqueswww** [comment]
Not really necessary, it's only parsing that needs to be checked ;)

**Comment by fubuloubu** [comment]
If both of those keywords did nothing in the compiler, it would still pass. Ergo, it should have a short test

File: tests/examples/safe_remote_purchase/test_safe_remote_purchase.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def received(): modifying
     def purchase(): modifying
     def unlocked() -> bool: constant
 
-purchase_contract: PurchaseContract
+purchase_contract: address(PurchaseContract)
```
Comment (fubuloubu) [inline tests/examples/safe_remote_purchase/test_safe_remote_purchase.py]: Undo this change

File: docs/index.rst [hunk 2]
```diff
@@ -55,6 +55,8 @@ Following the principles and goals, Vyper **does not** provide the following fea
 Compatibility-breaking Changelog
 ********************************
 
+* **2019.02.14**: Creating a persistent contract address can only be done using the _address_ syntax: `bar_contact: address(ERC20)`.
```
Comment (fubuloubu) [inline docs/index.rst]: Is this line still true?

**Comment by jacqueswww** [comment]
Will fix it thx ;)
